src/fetch/webfinger.rs

```diff
@@ -117,6 +117,75 @@ pub fn build_webfinger_response(subject: String, url: Url) -> Webfinger {
     }
 }
 
+/// Builds a webfinger response similar to `build_webfinger_response` that can have multiple
+/// actors and an optional type.
+///
+/// `urls` takes a vector of tuples. The first item of the tuple is the URL while the second
+/// item is the type, such as `"Person"` or `"Group"`. If `None` is passed for the type, the field
+/// will be empty.
+///
+/// ```
+/// # use url::Url;
+/// # use activitypub_federation::fetch::webfinger::build_webfinger_response_with_type;
+/// let subject = "acct:nutomic@lemmy.ml".to_string();
+/// let url = Url::parse("https://lemmy.ml/u/nutomic")?;
+/// build_webfinger_response_with_type(subject, vec![(url, None)]);
+/// # Ok::<(), anyhow::Error>(())
+/// ```
+///
+/// ```
+/// # use url::Url;
+/// # use activitypub_federation::fetch::webfinger::build_webfinger_response_with_type;
+/// let subject = "acct:nutomic@lemmy.ml".to_string();
+/// let url = Url::parse("https://lemmy.ml/u/nutomic")?;
+/// build_webfinger_response_with_type(subject, vec![(url, Some("Person"))]);
+/// # Ok::<(), anyhow::Error>(())
+/// ```
+/// ```
+/// # use url::Url;
+/// # use activitypub_federation::fetch::webfinger::build_webfinger_response_with_type;
+/// let subject = "acct:asklemmy@lemmy.ml".to_string();
+/// let url = Url::parse("https://lemmy.ml/c/asklemmy")?;
+/// build_webfinger_response_with_type(subject, vec![(url, Some("Group"))]);
+/// # Ok::<(), anyhow::Error>(())
+/// ```
+pub fn build_webfinger_response_with_type(
+    subject: String,
+    urls: Vec<(Url, Option<&str>)>,
```

Comment:
I don't like this URLs parameter much, its confusing. Maybe we could use a struct with derive_builder?

Reply:
Unfortunately I've never used `derive_builder` before. I tried to implement it for `Webfinger` and I ran into some problems. I was trying to use `#[builder(setter(each(name = "link")))]` where the `link` function would append a new URL and type to the links field. But it didn't work because the function signature is different. I also tried making a separate impl for `WebfingerBuilder` but the fields are `Option`s so it's a little difficult to do. I'm not sure how to proceed here since I'm trying to avoid messing with the struct itself. Any advice?

Reply:
Okay its fine then, maybe we can change it in the future.